### ISSUE TITLE
failed to recover from reboot.

### ISSUE THREAD

**Issue by shali-mor** [CiscoCloud/mantl#1238]
Hi , 
When trying to run the reboot script : ansible-playbook playbooks/reboot-hosts.yml
everything looks ok :
```
PLAY RECAP ******************************************************************** 
wait for host boot ----------------------------------------------------- 90.27s
set consul maintenance enable ------------------------------------------- 4.40s
wait for consul to listen ----------------------------------------------- 3.66s
reboot host ------------------------------------------------------------- 2.49s
set consul maintenance disable ------------------------------------------ 1.23s
shali-control-01           : ok=5    changed=3    unreachable=0    failed=0   
shali-control-02           : ok=5    changed=3    unreachable=0    failed=0   
shali-control-03           : ok=5    changed=3    unreachable=0    failed=0   
shali-edge-01              : ok=5    changed=3    unreachable=0    failed=0   
shali-edge-02              : ok=5    changed=3    unreachable=0    failed=0   
shali-worker-001           : ok=5    changed=3    unreachable=0    failed=0   
shali-worker-002           : ok=5    changed=3    unreachable=0    failed=0   
shali-worker-003           : ok=5    changed=3    unreachable=0    failed=0  
```
looking at mantl UI , it seems as it's in critical state (top-right it marked as critical) 

looking at mesos, it seems as if it has no resources (No master is currently leading ...
): 
I0307 09:34:18.661811  3272 contender.cpp:149] Joining the ZK group
I0307 09:34:28.397053  3283 recover.cpp:111] Unable to finish the recover protocol in 10secs, retrying

The marathon , and chronos appears as critical with no ability to access their UI (in marathon : Problem accessing /. Reason:
    Could not determine the current leader) 

I'm running on CCS . 
some information on my setup :

I'm using the following versions : 
Mantl 1.0.3-RC1
terraform : v0.6.11
Python 2.7.11
ansible 1.9.4

**Comment by darKoram** [comment]
terraform on aws generates instances with public ips.  If you reboot the instance, you get a new public ip.  The public ips should be converted to elastic ips to survive reboot.

**Comment by kbroughton** [comment]
possible related to https://github.com/CiscoCloud/mantl/issues/538 as well?

I tried inserting the "aws_eip" line in aws.tf, but strangely it only worked for the -01 edge, control and worker nodes.  I'm very new to terraform, but how is my block different from the "aws_volume_attachment" below the instances block (which does mound /dev/xvdh to all instances)

```
resource "aws_eip" "eip" {
    instance = "${element(aws_instance.instance.*.id, count.index)}"
    vpc = true
}

resource "aws_instance" "instance" {
  ami = "${var.source_ami}"
  instance_type = "${var.ec2_type}"
  count = "${var.count}"
  vpc_security_group_ids = [ "${split(",", var.security_group_ids)}"]
  key_name = "${var.ssh_key_pair}"
  associate_public_ip_address = true
  subnet_id = "${element(split(",", var.vpc_subnet_ids), count.index)}" 
  iam_instance_profile = "${var.iam_profile}"
  root_block_device {
    delete_on_termination = true
    volume_size = "${var.ebs_volume_size}"
  }


  tags {
    Name = "${var.short_name}-${var.role}-${format("%02d", count.index+1)}"
    sshUser = "${var.ssh_username}"
    role = "${var.role}"
    dc = "${var.datacenter}"
  }
}

resource "aws_volume_attachment" "instance-lvm-attachment" {
  count = "${var.count}"
  device_name = "xvdh"
  instance_id = "${element(aws_instance.instance.*.id, count.index)}"
  volume_id = "${element(aws_ebs_volume.ebs.*.id, count.index)}"
  force_detach = true
}
```

**Comment by siddharthist** [comment]
@darKoram @kbroughton It's not necessary to use eips, you can just run `terraform refresh`.

**Comment by sehqlr** [comment]
I just merged #1815, which should fix the root of this problem. I'm going to close issue for now; reopen if that PR didn't help. Thanks, all!
